packages/babel-plugin-transform-arrow-functions/test/fixtures/arrow-functions/arguments/output.js

```diff
@@ -81,3 +83,68 @@ function six(obj) {
 }
 
 six();
+
+var seven = function () {
+  var arguments = 1;
+  return arguments;
+};
+
+seven();
+
+var eight = function () {
+  var arguments = 1;
+  return function () {
+    return _arguments6;
+  };
```

Comment:
This looks wrong to me, since it doesn't return `1`.

Reply:
The rest looks fine?

Reply:
To fix this, when we find an `arguments` binding in the `do-while` loop, we should probably call `curr.rename("arguments")` so that it changes the binding name.